ros/src/computing/perception/detection/packages/road_wizard/nodes/feat_proj/feat_proj.cpp

```diff
@@ -270,44 +277,97 @@ double GetSignalAngleInCameraSystem(double hang, double vang)
 	return ConvertRadianToDegree(signal_pitch_in_cam);   // holizontal angle of camera is represented by pitch
 }  // double GetSignalAngleInCameraSystem()
 
-
-void echoSignals2(ros::Publisher &pub, bool useOpenGLCoord = false)
+Tbl read_csv(const char* filename, int* max_id)
 {
-	int countPoint = 0;
-	autoware_msgs::Signals signalsInFrame;
-
-	/* Get signals on the path if vecter_map_server is enabled */
-	if (g_use_vector_map_server)
-	{
-		vector_map_server::GetSignal service;
-		/* Set server's request */
-		service.request.pose = g_vector_map_client.pose();
-		service.request.waypoints = g_vector_map_client.waypoints();
+  std::ifstream ifs(filename);
+  std::string line;
+  std::getline(ifs, line); // Remove first line
+
+  Tbl tbl;
+
+  *max_id = -1;
+  while (std::getline(ifs, line)) {
+    std::istringstream ss(line);
+
+    std::vector<std::string> columns;
+    std::string column;
+    while (std::getline(ss, column, ',')) {
+      columns.push_back(column);
+    }
+    tbl.push_back(columns);
+
+    int id = std::stoi(columns[0]);
+    if (id > *max_id) *max_id = id;
```

Comment:
put this in {}

Reply:
fixed.

Reply:
Can you please add documentation on how to create the CSV?

A cleaner way to create the file would be only adding the `signal.id`. Other fields would be read directly from the VectorMap. Reducing the chance to errors.

Reply:
Yeah, it would be cleaner. But it would again be the same logic of fetching the signal information from looping through VectorMap based on signalId.

Reply:
but only once :)